src/augment-pr-entry/augment-pr-entry.js

```diff
@@ -74,16 +81,39 @@ export const addCreationDate = async (prNode, prData) => {
     prNumberAndTimestamp.appendChild(creationDateNode);
 };
 
+export const addUsernameWithLatestUpdate = async (prNode, prActivity) => {
+    // pull requests by default have the initial commit info as an activity
+    const mostRecentAction = prActivity.values[0];
+    let author = '';
+
+    // merges, commit updates
+    if (mostRecentAction.update) {
+        author = mostRecentAction.update.author.display_name;
+    } else if (mostRecentAction.approval) {
+        // approvals
+        author = mostRecentAction.approval.user.display_name;
+    } else if (mostRecentAction.comment) {
+        // comments
+        author = mostRecentAction.comment.user.display_name;
+    }
+
+    const prUpdateTime = prNode.querySelector('.pr-number-and-timestamp')
+        .firstElementChild;
+    prUpdateTime.append(` by ${author}`);
+};
+
 export default async function augmentPrEntry(prNode) {
```

Comment:
The body of this function could be improved. The way the code is laid out right now with the `await`s, we are blocking the firing of the next HTTP request until the first one is completed, and then blocking each step of the DOM manipulation. 

This is not actually necessary becuase each execution flow is independent from the other and they could do their work on their own context without interfering with each other. Doing it like that would produce a faster load time for the user! It would look like this:

```js
export default function augmentPrEntry(prNode) {
    linkifyTargetBranch(prNode);

    const prId = prNode.dataset.pullRequestId;

    getPrData(prId).then(prData => {
        if (prData) {
            addSourceBranch(prNode, prData);
            addCreationDate(prNode, prData);
        }
    });

    getPrActivity(prId).then(prActivity => {
        if (prActivity) {
            addUsernameWithLatestUpdate(prNode, prActivity);
        }
    });
}
```

I dropped the `await`s entirely because they basically make the body of the function synchronous, which is not what we would want here. If we wanted to keep the `await`s, we would have to introduce more functions, bloating more the file. Good ol' `.then`s would do the job just fine! :D

Reply:
Sweet. so true! 😄